gateway/src/model.rs

```diff
@@ -106,6 +106,20 @@ impl ModelConfig {
         }
         Err(Error::ModelProvidersExhausted { provider_errors })
     }
+
+    pub fn validate(&self) -> Result<(), Error> {
+        // Ensure that at least one provider has credentials
+        if !self
+            .providers
+            .values()
+            .any(|provider| provider.has_credentials())
```

Comment:
Maybe this should be a `all`? Otherwise we might accidentally always trigger a fallback unknowingly to the user

Reply:
it will log, I think we should always do best-effort and in that case it would be a fallback that is in observability and logs both,